gcc/rust/rust-diagnostics.cc

```diff
@@ -210,18 +213,10 @@ class rust_error_code_rule : public diagnostic_metadata::rule
   char *make_description () const final override
   {
     // 'E' + 4 characters + \0
-    char *buffer = new char[6];
-
-    // is that needed. does C++ suck that much that you
-    // can't zero initialize a new[]'d char array
-    memset (buffer, 0, 6);
+    char *buffer = (char *) xcalloc (6, sizeof (char));
```

Comment:
I would have used a `static_cast` as most people recommend it.

Reply:
Fixed in 7be3d0817ad